sound/soc/sof/intel/hda-codec.c

```diff
@@ -131,10 +130,16 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address)
 	    (resp & 0xFFFF0000) != IDISP_VID_INTEL) {
 		hdev->type = HDA_DEV_LEGACY;
 		hda->hda_codec_mask |= BIT(address);
-		hda_codec_load_module(&hda_priv->codec);
+		ret = hda_codec_load_module(&hda_priv->codec);
+		/*
+		 * handle ret==0 (no driver bound) as an error, but pass
+		 * other return codes without modification
+		 */
+		if (ret == 0)
```

Comment:
I think checkpatch suggests to use `if (!ret)` in such cases

Reply:
@lyakh I had that in the original version, but as the return value is a more complex enumeration (both positive and negative values have specific meanings), I wanted to highlight zero on purpose.
Original review to sof-dev is at: https://github.com/thesofproject/linux/pull/1679